internal/api/resolver_mutation_scene.go

```diff
@@ -643,6 +643,176 @@ func (r *mutationResolver) SceneMerge(ctx context.Context, input SceneMergeInput
+		Contrast:    input.Contrast,
+		Brightness:  input.Brightness,
+		Gamma:       input.Gamma,
+		Saturate:    input.Saturate,
+		HueRotate:   input.HueRotate,
+		Warmth:      input.Warmth,
+		Red:         input.Red,
+		Green:       input.Green,
+		Blue:        input.Blue,
+		Blur:        input.Blur,
+		Rotate:      input.Rotate,
+		Scale:       input.Scale,
+		AspectRatio: input.AspectRatio,
+		SceneID:     sceneID,
+		CreatedAt:   currentTime,
+		UpdatedAt:   currentTime,
+	}
+
+	err = r.changeFilter(ctx, create, &newSceneFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	r.hookExecutor.ExecutePostHooks(ctx, newSceneFilter.ID, plugin.SceneFilterCreatePost, input, nil)
+	return r.getSceneFilter(ctx, newSceneFilter.ID)
+}
+
+func (r *mutationResolver) SceneFilterUpdate(ctx context.Context, input SceneFilterUpdateInput) (*models.SceneFilter, error) {
+	// Populate scene filter from the input
+	sceneFilterID, err := strconv.Atoi(input.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	sceneID, err := strconv.Atoi(input.SceneID)
+	if err != nil {
+		return nil, err
+	}
+
+	updatedSceneFilter := models.SceneFilter{
+		ID:          sceneFilterID,
+		SceneID:     sceneID,
+		Contrast:    input.Contrast,
+		Brightness:  input.Brightness,
+		Gamma:       input.Gamma,
+		Saturate:    input.Saturate,
+		HueRotate:   input.HueRotate,
+		Warmth:      input.Warmth,
+		Red:         input.Red,
+		Green:       input.Green,
+		Blue:        input.Blue,
+		Blur:        input.Blur,
+		Rotate:      input.Rotate,
+		Scale:       input.Scale,
+		AspectRatio: input.AspectRatio,
+		UpdatedAt:   time.Now(),
+	}
+
+	err = r.changeFilter(ctx, update, &updatedSceneFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	translator := changesetTranslator{
+		inputMap: getUpdateInputMap(ctx),
+	}
+	r.hookExecutor.ExecutePostHooks(ctx, updatedSceneFilter.ID, plugin.SceneFilterUpdatePost, input, translator.getFields())
+	return r.getSceneFilter(ctx, updatedSceneFilter.ID)
+}
+
+func (r *mutationResolver) SceneFilterDestroy(ctx context.Context, id string) (bool, error) {
+	filterID, err := strconv.Atoi(id)
+	if err != nil {
+		return false, err
+	}
+
+	if err := r.withTxn(ctx, func(ctx context.Context) error {
+		qb := r.repository.SceneFilter
+		sqb := r.repository.Scene
+
+		filter, err := qb.Find(ctx, filterID)
+
+		if err != nil {
+			return err
+		}
+
+		if filter == nil {
+			return fmt.Errorf("scene filter with id %d not found", filterID)
+		}
+
+		s, err := sqb.Find(ctx, filter.SceneID)
+		if err != nil {
+			return err
+		}
+
+		if s == nil {
+			return fmt.Errorf("scene with id %d not found", filter.SceneID)
+		}
+
+		return scene.DestroyFilter(ctx, s, filter, qb)
+	}); err != nil {
+		return false, err
+	}
+
+	r.hookExecutor.ExecutePostHooks(ctx, filterID, plugin.SceneFilterDestroyPost, id, nil)
+
+	return true, nil
+}
+
+func (r *mutationResolver) changeFilter(ctx context.Context, changeType int, changedFilter *models.SceneFilter) error {
```

Comment:
This will need to be refactored to remove the `changeFilter` method, like I did to `changeMarker` in #3945 - the changeType "enum" no longer exists.